server/templates/obslist.html

```diff
@@ -21,14 +73,111 @@ <h3>Observations</h3>
             <td>{{ obs.los }}</td>
-            <td><a href="https://www.n2yo.com/satellite/?s={{ obs.sat_id }}">{{ obs.sat_name }}</a></td>
-            <td><a href="/station/{{ obs.station_id }}">{{ obs.station_id}}</a></td>
-            <td><a href="/obs/{{ obs.obs_id }}"> <img src="/data/thumbs/{{ obs.thumbnail }}"></a></td>
+            <td>
+                <a href="https://www.n2yo.com/satellite/?s={{ obs.sat_id }}">
+                    {{ obs.sat_name }}
+                </a>
+            </td>
+            <td>
+                <a href="/station/{{ obs.station_id }}">
+                    {{ obs.station_name }}
+                </a>
+            </td>
+            <td>
+                <a href="/obs/{{ obs.obs_id }}">
+                    <img src="/data/thumbs/{{ obs.thumbnail }}" />
+                </a>
+            </td>
         </tr>
+    {% else %}
+    <tr>
+        <td class="text-center" colspan="100%">No data for provided criteria</td>
+    </tr>
     {% endfor %}
 </table>
 </p>
 {% include 'pagination.html' %}
 
 {% endblock %}
 
+{% block scripts %}
+<script defer>
+    const form = document.getElementById("search-form");
+
+    function getFormControls()
+    {
+        return form.querySelectorAll("input, select");
+    }
+
+    function disableNonEmptyControls()
+    {
+        const controls = getFormControls();
+        for (let i=0; i < controls.length; i++)
+        {
+            const control = controls[i];
+            if (control.value == "") {
+                control.disabled = true;
+            }
+        }
+    }
+
+    function search_for_parent(element, className, maximum_parent) {
+        while (element.parentElement != null && element != maximum_parent)
+        {
+            if (element.classList.contains(className))
+            {
+                return element;
+            }
+            element = element.parentElement;
+        }
+        return null;
+    }
+
+    function validate()
+    {
+        let is_valid = true;
+        const error_control_containers = form.querySelectorAll(".has-error");
+        error_control_containers.forEach(c => c.classList.remove("has-error"));
+
+        const from_input = form.querySelector("input[name=los_after]");
+        const to_input = form.querySelector("input[name=aos_before]");
+        
+        if (from_input.value != "" && to_input.value != "") {
+            const from_date = new Date(from_input.value);
+            const to_date = new Date(to_input.value);
+            if (from_date > to_date)
+            {
+                is_valid = false;
+                [from_input, to_input].forEach(input => {
+                    parent = search_for_parent(input, "form-group", form);
+                    parent.classList.add("has-error")
+                });
+            }
+        }
+
+        return is_valid;
+    }
+
+    form.onsubmit = function(ev) {
+        if (!validate())
+        {
+            ev.preventDefault();
```

Comment:
What does preventDefault() do?

Reply:
```preventDefault``` tells the browser that should suppress default handling of this event.

Standard behavior of forms in HTML is send request to target address. ```onSubmit``` handler is called before sending.  When validation fails then we stop submit using ```preventDefault```.

Reply:
Thanks for the explanation. Can you put as a comment?

Reply:
```suggestion
            ev.preventDefault(); // prevent default event handling by a browser.
```